### DISCUSSION TITLE
STATA on level 4 server

### DISCUSSION THREAD

**Discussion by NikNakk** [opensafely/documentation#240] [Q&A] [answered]
I understand that some users have access to STATA on the level 4 TPP server in order to preview outputs to see if they need redaction. How can I gain access to this? Note I already have access to the TPP VPN and Level 4 server, but cannot see STATA installed there.

**Comment by acagreen17** [comment, marked as answer]
Hi Nick,

I'm afraid there is no STATA (or any other of the common statistical software that one might use for analysis) install on the TPP server. To preview outputs to see if they need redaction, you'll need to save your outputs as one of the following file types; PDF; HTML; TXT; CSV; SVG; JPG; (Please contact us if you would like additional file types added, we are happy to review).

For further information on output checking, see the [Disclosivity checks and redaction](https://docs.opensafely.org/releasing-files/#disclosivity-checks-and-redaction) section in the onboarding documentation. 

I hope that answers your question?

**Reply by NikNakk** [reply]
Thanks, I must have misunderstood what my colleague said. Is it possible to get pandas installed into the Python 3 installation that’s already there? It would make examining Stata .dta files possible, but would also make it easier to check csvs.

**Reply by remlapmot** [reply]
Just to add that it is helpful to save your Stata log files as text files (i.e. not in SMCL format) then you can more easily view them in a text editor (admittedly SMCL files are text files just with some markup), 
```stata
log using mylogfile.log, text replace
```
e.g. an example [here](https://github.com/opensafely/ethnicity-covid-research/blob/b2a29e8b9a99092bea2d860612b4a07b29be410b/analysis/00_eth_flowchart.do#L18)

**Reply by NikNakk** [reply]
Thanks. We’ve already done so, but they’re not that easy to parse (as a human) because of the markup, and I’ve also been asked to release the .dta files which I’d therefore also like to check prior to release. If that’s not possible, I may have to just request release of the logs.

**Reply by alexwalkerepi** [reply]
Hi Nick, the `.dta` output tables will need to be exported as CSVs at the moment. There are plans to add software to make it easier to view CSVs soon.

**Comment by NikNakk** [comment]
Thanks. Any prospect of adding pandas to Python?

Nick

**Reply by acagreen17** [reply]
Morning Nick, it's unlikely that pandas will be added to Python any time soon. Can I ask what it is that you need to see from the .dta files? i.e., is it an output form a model/table of results? If you let us know what the outputs are that you're trying to review we can have a think about what might be the best way for you to view them in the short term. And in the long term, it will help us when we come to building output viewers.

**Reply by NikNakk** [reply]
Hi Millie,

STATA isn’t my language of choice (I use R and a bit of Python), but some of my collaborators use STATA and were keen for me to export the .dta files. I wanted to be able to check them for any possibly disclosive data before doing so.

It looks like that won’t be possible at this stage, so we’ll work around it. I thought that since Python was already in the level 4 environment, and pandas is a widely-used package, that might be a solution, but I of course understand the need to keep the environment as minimal as possible.
